src/biogeochem/FireEmisFactorsMod.F90

```diff
@@ -73,7 +74,9 @@ subroutine fire_emis_factors_get( comp_name, factors, molecwght )
        call endrun(errmes)
     endif
 
-    factors(:npfts) = comp_factors_table( ndx )%eff(:npfts)
+    factors(:maxveg) = comp_factors_table( ndx )%eff(:maxveg)
+    ! If fire emissions factor file only includes natural PFT's, but this is a crop case
+    ! Copy the generic crop factors to the crop CFT's from generic crop
     if ( size(factors) > npfts )then
```

Comment:
I don't think this can be reached, given the error check at line 132. That's why I deleted this block in my patch.

Reply:
Yeah, I'm debugging to get this to work now, as I think it might be useful to be able to use the old 16-pft fire emission dataset.

Reply:
Got this working. So resolving.

Reply:
Well my original concern is still there—`if ( size(factors) > npfts )` will never be true here as long as `if ( npfts < maxveg )` in `fire_emis_factors_init()` triggers an error.

Do we want the user to be able to run with a too-short emissions file or not? If yes, delete that error check (and rename `npfts` to something more descriptive, like `npfts_on_emis_file`). If no, delete this conditional (and the `npft` variable). 

I say no—I don't like it when the model silently resolves what might have been a mistake on my part.

Reply:
For the "maybe" use-case where the user wants to use the old 16-member emission factor dataset, they can fill out the other 62 PFTs with dummy data. Or maybe we could introduce a namelist option, but I think that's more trouble than it's worth.

Reply:
It's a fair question to ask if we should trigger an error if you give an old 16-PFT fire emission dataset. I'm open to either. But, we'll make this something we resolve post ctsm5.3.0. So I'll elevate this to an issue and add the next label to it.

Reply:
Resolving this and we will take care of it in #2781